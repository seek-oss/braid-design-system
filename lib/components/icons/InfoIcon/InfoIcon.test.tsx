import React from 'react';
import { InfoIcon } from '../..';
import { render } from 'react-testing-library';
import forEachTheme from '../../../test/utils/forEachTheme';

describe('InfoIcon', () => {
  forEachTheme(({ ThemeProvider }) => {
    test('Default', () => {
      const { container } = render(
        <ThemeProvider>
          <InfoIcon />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Sized', () => {
      const { container } = render(
        <ThemeProvider>
          <InfoIcon size="large" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Inline', () => {
      const { container } = render(
        <ThemeProvider>
          <InfoIcon inline />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
