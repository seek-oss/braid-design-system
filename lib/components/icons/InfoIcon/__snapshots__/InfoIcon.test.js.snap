// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoIcon Theme: jobStreet Default 1`] = `
<svg
  class="Box__root borderColor__default"
  height="20"
  viewBox="0 0 30 30"
  width="20"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: jobStreet Inline 1`] = `
<svg
  class="Box__root borderColor__default"
  height="14"
  viewBox="0 0 30 30"
  width="14"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: jobStreet Sized 1`] = `
<svg
  class="Box__root borderColor__default"
  height="24"
  viewBox="0 0 30 30"
  width="24"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: seekAnz Default 1`] = `
<svg
  class="Box__root borderColor__default"
  height="24"
  viewBox="0 0 30 30"
  width="24"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: seekAnz Inline 1`] = `
<svg
  class="Box__root borderColor__default"
  height="16"
  viewBox="0 0 30 30"
  width="16"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: seekAnz Sized 1`] = `
<svg
  class="Box__root borderColor__default"
  height="24"
  viewBox="0 0 30 30"
  width="24"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: seekAsia Default 1`] = `
<svg
  class="Box__root borderColor__default"
  height="20"
  viewBox="0 0 30 30"
  width="20"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: seekAsia Inline 1`] = `
<svg
  class="Box__root borderColor__default"
  height="16"
  viewBox="0 0 30 30"
  width="16"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: seekAsia Sized 1`] = `
<svg
  class="Box__root borderColor__default"
  height="24"
  viewBox="0 0 30 30"
  width="24"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: wireframe Default 1`] = `
<svg
  class="Box__root borderColor__default"
  height="24"
  viewBox="0 0 30 30"
  width="24"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: wireframe Inline 1`] = `
<svg
  class="Box__root borderColor__default"
  height="16"
  viewBox="0 0 30 30"
  width="16"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;

exports[`InfoIcon Theme: wireframe Sized 1`] = `
<svg
  class="Box__root borderColor__default"
  height="24"
  viewBox="0 0 30 30"
  width="24"
  x="0"
  y="0"
>
  <path
    d="M14.905,21.843c-0.552,0-1-0.447-1-1v-8.236c0-0.552,0.448-1,1-1s1,0.448,1,1v8.236 C15.905,21.396,15.458,21.843,14.905,21.843z"
  />
  <path
    d="M14.905,10.208c-0.552,0-1-0.448-1-1s0.448-1.025,1-1.025s1,0.422,1,0.975v0.051 C15.905,9.76,15.458,10.208,14.905,10.208z"
  />
  <path
    d="M15.001,28.499C7.556,28.499,1.499,22.443,1.499,15c0-7.444,6.057-13.5,13.502-13.5 c7.442,0,13.498,6.056,13.498,13.5C28.499,22.443,22.443,28.499,15.001,28.499z M15.001,3.501C8.659,3.501,3.499,8.66,3.499,15 c0,6.34,5.16,11.499,11.502,11.499c6.34,0,11.498-5.158,11.498-11.499C26.499,8.66,21.341,3.501,15.001,3.501z"
  />
</svg>
`;
